gensim/models/word2vec.py

```diff
@@ -1112,7 +1112,7 @@ def update_weights(self):
         # randomize the remaining words
         for i in xrange(len(self.wv.syn0), len(self.wv.vocab)):
             # construct deterministic seed from word AND seed argument
-            newsyn0[i-len(self.wv.syn0)] = self.seeded_vector(self.wv.index2word[i] + str(self.seed))
+            newsyn0[i-len(self.wv.syn0)] = self.seeded_vector(utils.to_unicode(self.wv.index2word[i]) + str(self.seed))
```

Comment:
I thought about the same, though it seemed nice to be explicit about this. But you're right - once the problem in the corpus is addressed, this is not causing issues anymore.

I've reverted the changes.